src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -96,13 +96,16 @@ public void ValidateConfig()
             IFileSystem fileSystem,
             ILogger logger)
         {
-            // Connection string can't be null or empty
-            if (string.IsNullOrWhiteSpace(runtimeConfig.DataSource.ConnectionString))
+            foreach (DataSource dataSource in runtimeConfig.DatasourceNameToDataSource.Values)
             {
-                throw new DataApiBuilderException(
-                    message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                // Connection string can't be null or empty
+                if (string.IsNullOrEmpty(dataSource.ConnectionString))
```

Comment:
Will the data source names be uniquely generated by us in a multi-db scenario? If so, I assume they are for internal use to DAB only and should not be included in the exception message. I am thinking about the scenario where we have a validation error in one of the config files, and just by looking at the exception, we want to know which config file has the error.

Reply:
They will be generated by us internally so the customer will not be aware of them and ideally we will tie this to the filename when we do the merge. However, its a little tricky - We will have to validate this in the cli when we do the merge if that makes sense. To the core system there is just a single runtimeconfig file not multiple. Hence will do this validation, when I do the cli changes.

Reply:
I see. Thanks for the clarification.